### DISCUSSION TITLE
Next/image: How to keep the aspect ratio?

### DISCUSSION THREAD

**Discussion by madzadev** [vercel/next.js#22807] [Help] [answered]
I'm loading images via external CDN and using the Image component.

According to [docs](https://nextjs.org/docs/api-reference/next/image), `width` and `height` are required props. Same for the parent (when using `layout` prop).
In the above methods, images lose their aspect ratios due to being stretched to prop values.

Surely, it could work well when dealing with just a single image, or images with the same aspect ratio already from the source. Tho, in my case, I'm dealing with lots of images with a variety of resolutions in landscape and portrait orientations.

Is there a way to provide just the `width`, so that:
1.Images with smaller width are displayed in original res;
2.Images with higher width get resized to the specified width while keeping the aspect ratio?

**Comment by tvojtech** [comment]
If you don't care different image heights, you can use `layout="fill" objectFit="contain"`. 

If you'd like to create something like filmstrip (same height, different width according to aspect ratio, same space between images) I didn't find any way to do it for images with different ratios.

**Comment by akellbl4** [comment, marked as answer]
You have two options:

1. Provide `objectFit="cover"` plus `width` and `height` – your image will be scaled by width and placed in container with that size.
2. Provide `objectFit="cover"` with `layout="fill"` – in this case your images won't be scaled but optimized.
